$OpenBSD: patch-src_main_main_c,v 1.6 2014/03/12 11:04:41 dcoppa Exp $

OpenBSD always returns EINVAL when trying to disable an active stack

--- src/main/main.c.orig	Tue Jul 16 00:05:04 2013
+++ src/main/main.c	Tue Feb  4 09:15:53 2014
@@ -28,6 +28,8 @@
 #include <ctype.h>
 #include <stdlib.h>
 #include <string.h>
+#include <errno.h>
+#include <sys/errno.h>
 
 #define __MAIN__
 #define R_USE_SIGNALS 1
@@ -619,7 +621,8 @@ static void init_signal_handlers(void)
 	sigstk.ss_size = SIGSTKSZ + R_USAGE;
 	sigstk.ss_flags = 0;
 	if(sigaltstack(&sigstk, NULL) < 0)
-	    warning("failed to set alternate signal stack");
+	    if (errno != EINVAL) /* OpenBSD always returns EINVAL */
+	    	warning("failed to set alternate signal stack");
     } else
 	warning("failed to allocate alternate signal stack");
     sa.sa_sigaction = sigactionSegv;
